include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -382,6 +355,24 @@ inline void* __attribute__((always_inline)) __libc_realloc(void* __ptr, std::siz
     return realloc(__ptr, __size);
 }
 
+inline void* __attribute__((always_inline)) valloc(std::size_t __size)
+{
+    return memalign(__pstl_offload::__get_memory_page_size(), __size);
+}
+
+inline void* __attribute__((always_inline)) __libc_valloc(std::size_t __size) { return valloc(__size); }
+
+// __THROW to match system declaration of pvalloc
+inline void* __attribute__((always_inline)) pvalloc(std::size_t __size) __THROW
+{
+    size_t __page_size = __pstl_offload::__get_memory_page_size();
+    // align size up to the page size
+    __size = __size ? ((__size - 1) | (__page_size - 1)) + 1 : __page_size;
```

Comment:
I don't understand `size ?` part. Can our `__get_memory_page_size` return `0`?

Reply:
> I don't understand size ? part. 
We mimic behavior of replaced `pvalloc()` . 

> Can our __get_memory_page_size return 0?

I'm afraid I'm not understand the question. Are you asking what we will do if somehow a system call returns completely invalid value? 

Just in case I will add checking of `sysconf()` return value.

Reply:
yep, my bad, I misread the variable names